### ISSUE TITLE
make test fails for all sketch types

### ISSUE THREAD

**Issue by nikunjbhartia** [apache/datasketches-bigquery#77]
Getting different errors for different tests. 

Steps: 
```shell
export BQ_PROJECT='nikunjbhartia-test-clients'
export BQ_DATASET='test_us'
export JS_BUCKET='nikunj_test_us'

> Make all 
> Make install 
make test on each sketch directory
```

Output: 
Theta
```shell
_/datasketches-bigquery/theta (main)
->make test
Waiting on bqjob_r46ee75b147317e40_00000192a86cb86d_1 ... (7s) Current status: DONE   
Error in query string: Error processing job 'nikunjbhartia-test-clients:bqjob_r46ee75b147317e40_00000192a86cb86d_1': Query error: Error: sketch type mismatch: expected 3, actual 81 at UDF$2(BYTES, INT64,
INT64) line 12, columns 2-3 at [29:1]
make: *** [test] Error 1
``` 

KLL
```shell
 datasketches-bigquery/kll (main)
->make test
Waiting on bqjob_r67dd3f36e807bbae_00000192a86d037d_1 ... (4s) Current status: DONE   
Error in query string: Error processing job 'nikunjbhartia-test-clients:bqjob_r67dd3f36e807bbae_00000192a86d037d_1': Query error: ReferenceError: sketch is not defined at
test_us.kll_sketch_float_build(FLOAT64, INT64) line 44, columns 27-28 at [31:1]
Waiting on bqjob_r177315ed6dda367e_00000192a86d1ed2_1 ... (5s) Current status: DONE   
Error in query string: Error processing job 'nikunjbhartia-test-clients:bqjob_r177315ed6dda367e_00000192a86d1ed2_1': Query error: Error: Possible corruption: M must be 8: 103 at UDF$1(BYTES) line 12, columns
2-3 at [27:1]
make: *** [test] Error 1
```

CPC
```shell
datasketches-bigquery/cpc (main)
->make test
Waiting on bqjob_r229098308cdf4c06_00000192a86d7a2f_1 ... (1s) Current status: DONE   
Error in query string: Error processing job 'nikunjbhartia-test-clients:bqjob_r229098308cdf4c06_00000192a86d7a2f_1': Query error: Error: Insufficient buffer size detected: bytes available 56, minimum needed
268 at UDF$3(BYTES, BYTES, INT64, INT64) line 8, columns 2-3 at [22:1]
make: *** [test] Error 1
```

Tuple
```shell
datasketches-bigquery/tuple (main)
->make test
Waiting on bqjob_r10c20c5743c157b7_00000192a86da64e_1 ... (5s) Current status: DONE   
Error in query string: Error processing job 'nikunjbhartia-test-clients:bqjob_r10c20c5743c157b7_00000192a86da64e_1': Query error: Error: serial version mismatch: expected 3 or 1, actual 119 at UDF$1(BYTES,
INT64, INT64, INT64) line 7, columns 2-3 at [931:1]
Waiting on bqjob_r4032e6b6333e6dd_00000192a86dc779_1 ... (1s) Current status: DONE   
Error in query string: Error processing job 'nikunjbhartia-test-clients:bqjob_r4032e6b6333e6dd_00000192a86dc779_1': Query error: Error: sketch type mismatch: expected 3, actual 81 at UDF$2(BYTES, INT64,
INT64) line 6, columns 2-3 at [23:1]
make: *** [test] Error 1
```

HLL
```shell
datasketches-bigquery/hll (main)
->make test
Waiting on bqjob_r76d2320aa4aef686_00000192a86e09f3_1 ... (0s) Current status: DONE   
Error in query string: Error processing job 'nikunjbhartia-test-clients:bqjob_r76d2320aa4aef686_00000192a86e09f3_1': Query error: Error: Attempt to deserialize unknown object type at UDF$2(BYTES) line 6,
columns 2-3 at [21:1]
make: *** [test] Error 1
```

FI
```shell
datasketches-bigquery/fi (main)
->make test
Waiting on bqjob_r5a1b94eb93bce955_00000192a86e26f1_1 ... (0s) Current status: DONE   
Error in query string: Error processing job 'nikunjbhartia-test-clients:bqjob_r5a1b94eb93bce955_00000192a86e26f1_1': Query error: Error: Possible corruption: preamble longs of an empty sketch must be 1: 66 at
UDF$2(BYTES) line 6, columns 2-3 at [20:1]
make: *** [test] Error 1
```

**Comment by jmalkin** [comment]
The errors seem like they could be base64 vs binary encoding?

**Comment by nikunjbhartia** [comment]
yes, you are right, when I tried to deploy these functions in another project in which the UDF feature is allowlisted, i get following errors which are same: ( not sure what am I missing here)

```shell
->make test
/Library/Developer/CommandLineTools/usr/bin/make -C theta test # Added '-' to continue on error
Waiting on bqjob_r371e3eabacf4821f_00000192afd69c09_1 ... (7s) Current status: DONE   
Error in query string: Error processing job 'bigquerytestdefault:bqjob_r371e3eabacf4821f_00000192afd69c09_1': Query error: Error: sketch type mismatch: expected 3, actual 81 at UDF$2(BYTES, INT64, INT64) line
12, columns 2-3 at [29:1]
make[1]: *** [test] Error 1
make: [theta.test] Error 2 (ignored)

/Library/Developer/CommandLineTools/usr/bin/make -C tuple test # Added '-' to continue on error
Waiting on bqjob_r55b31d226e8a5189_00000192afd6c283_1 ... (5s) Current status: DONE   
Error in query string: Error processing job 'bigquerytestdefault:bqjob_r55b31d226e8a5189_00000192afd6c283_1': Query error: Error: serial version mismatch: expected 3 or 1, actual 119 at UDF$1(BYTES, INT64,
INT64, INT64) line 7, columns 2-3 at [931:1]
Waiting on bqjob_r4882d35977166fb1_00000192afd6e370_1 ... (2s) Current status: DONE   
Error in query string: Error processing job 'bigquerytestdefault:bqjob_r4882d35977166fb1_00000192afd6e370_1': Query error: Error: sketch type mismatch: expected 3, actual 81 at UDF$2(BYTES, INT64, INT64) line
6, columns 2-3 at [23:1]
make[1]: *** [test] Error 1
make: [tuple.test] Error 2 (ignored)

/Library/Developer/CommandLineTools/usr/bin/make -C cpc test # Added '-' to continue on error
Waiting on bqjob_r680f1c32fcbeb6c2_00000192afd6f5f5_1 ... (1s) Current status: DONE   
Error in query string: Error processing job 'bigquerytestdefault:bqjob_r680f1c32fcbeb6c2_00000192afd6f5f5_1': Query error: Error: Insufficient buffer size detected: bytes available 56, minimum needed 268 at
UDF$3(BYTES, BYTES, INT64, INT64) line 8, columns 2-3 at [22:1]
make[1]: *** [test] Error 1
make: [cpc.test] Error 2 (ignored)

/Library/Developer/CommandLineTools/usr/bin/make -C hll test # Added '-' to continue on error
Waiting on bqjob_r7d1ceae551a047eb_00000192afd70678_1 ... (0s) Current status: DONE   
Error in query string: Error processing job 'bigquerytestdefault:bqjob_r7d1ceae551a047eb_00000192afd70678_1': Query error: Error: Attempt to deserialize unknown object type at UDF$2(BYTES) line 6, columns 2-3
at [21:1]
make[1]: *** [test] Error 1
make: [hll.test] Error 2 (ignored)

/Library/Developer/CommandLineTools/usr/bin/make -C kll test # Added '-' to continue on error
Waiting on bqjob_r2b20176901cb03bc_00000192afd711f6_1 ... (5s) Current status: DONE   
Error in query string: Error processing job 'bigquerytestdefault:bqjob_r2b20176901cb03bc_00000192afd711f6_1': Query error: ReferenceError: sketch is not defined at
000_nikunj_us.kll_sketch_float_build(FLOAT64, INT64) line 44, columns 27-28 at [31:1]
Waiting on bqjob_r743b87592d94f7dc_00000192afd732d4_1 ... (7s) Current status: DONE   
Error in query string: Error processing job 'bigquerytestdefault:bqjob_r743b87592d94f7dc_00000192afd732d4_1': Query error: Error: Possible corruption: M must be 8: 103 at UDF$1(BYTES) line 12, columns 2-3 at
[27:1]
make[1]: *** [test] Error 1
make: [kll.test] Error 2 (ignored)

/Library/Developer/CommandLineTools/usr/bin/make -C fi test # Added '-' to continue on error
Waiting on bqjob_r66677fba22dc05e7_00000192afd75a03_1 ... (0s) Current status: DONE   
Error in query string: Error processing job 'bigquerytestdefault:bqjob_r66677fba22dc05e7_00000192afd75a03_1': Query error: Error: Possible corruption: preamble longs of an empty sketch must be 1: 66 at
UDF$2(BYTES) line 6, columns 2-3 at [20:1]
make[1]: *** [test] Error 1
make: [fi.test] Error 2 (ignored)
./tests/run_dataform_tests.sh
Compiling...

Compiled successfully.

```

**Comment by AlexanderSaydakov** [comment]
This seems to be because of encoding.
Do you have the new feature js_parameter_encoding_mode='STANDARD' enabled for you?

**Comment by nikunjbhartia** [comment]
The `bigquerytestdefault` that you see in the job id is allowlisted for the feature `js_parameter_encoding_mode`
Are there more features which were allowlisted for you in the process ?

**Comment by AlexanderSaydakov** [comment]
I am not aware of any other features.

**Comment by nikunjbhartia** [comment]
The issue is fixed after I tested on my project which is now allowlisted. Thanks ! 

There is another issue due to the order in which functions are created which lead to creation failure of some of the functions  during `make install`

```
$/datasketches-bigquery/kll 
->make install
...
Waiting on bqjob_r140c9a74346a3922_00000192f803cd27_1 ... (0s) Current status: DONE   
Error in query string: Error processing job 'nikunjbhartia-test-clients:bqjob_r140c9a74346a3922_00000192f803cd27_1': Function not found: `nikunjbhartia-test-clients.test_us.kll_sketch_float_build_k` at [34:3]
Failure details:
- query: Function not found: `nikunjbhartia-test-
clients.test_us.kll_sketch_float_build_k` at [34:3]
creating sqlx/kll_sketch_float_build_k.sqlx
Waiting on bqjob_r60d214316af87752_00000192f803d526_1 ... (0s) Current status: DONE   
Created nikunjbhartia-test-clients.test_us.kll_sketch_float_build_k
...
creating sqlx/kll_sketch_float_merge.sqlx
Waiting on bqjob_r6fb45bff6329b587_00000192f80421de_1 ... (0s) Current status: DONE   
Error in query string: Error processing job 'nikunjbhartia-test-clients:bqjob_r6fb45bff6329b587_00000192f80421de_1': Function not found: `nikunjbhartia-test-clients.test_us.kll_sketch_float_merge_k` at [34:3]
Failure details:
- query: Function not found: `nikunjbhartia-test-
clients.test_us.kll_sketch_float_merge_k` at [34:3]
creating sqlx/kll_sketch_float_merge_k.sqlx
Waiting on bqjob_r4a18f38ec91591df_00000192f8042851_1 ... (0s) Current status: DONE   
Created nikunjbhartia-test-clients.test_us.kll_sketch_float_merge_k
...
```

tdigest: 
```
...
creating sqlx/tdigest_double_build.sqlx
Waiting on bqjob_r73ad65ba64b73c1b_00000192f80a49e7_1 ... (0s) Current status: DONE   
Error in query string: Error processing job 'nikunjbhartia-test-clients:bqjob_r73ad65ba64b73c1b_00000192f80a49e7_1': Function not found: `nikunjbhartia-test-clients.test_us.tdigest_double_build_k` at [34:3]
Failure details:
- query: Function not found: `nikunjbhartia-test-
clients.test_us.tdigest_double_build_k` at [34:3]
creating sqlx/tdigest_double_build_k.sqlx
...
creating sqlx/tdigest_double_merge.sqlx
Waiting on bqjob_ra71dbfa2b6bdd15_00000192f80a78bc_1 ... (0s) Current status: DONE   
Error in query string: Error processing job 'nikunjbhartia-test-clients:bqjob_ra71dbfa2b6bdd15_00000192f80a78bc_1': Function not found: `nikunjbhartia-test-clients.test_us.tdigest_double_merge_k` at [34:3]
Failure details:
- query: Function not found: `nikunjbhartia-test-
clients.test_us.tdigest_double_merge_k` at [34:3]
creating sqlx/tdigest_double_merge_k.sqlx
Waiting on bqjob_r6fe1676dcee9cc8b_00000192f80a7f68_1 ... (0s) Current status: DONE   
...
```

Note that this is not a blocker for integration with bqutils repo as function creation there is handled using dataform which maintains a lineage of sqlx definitions and manages creation order accordingly.

**Comment by AlexanderSaydakov** [comment]
the creation order should be fixed by #88

**Comment by nikunjbhartia** [comment]
This now works as intended. Closing this ticket. Thanks.
